src/server/journal/serializer.cc

```diff
@@ -0,0 +1,204 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+//
+
+#include "server/journal/serializer.h"
+
+#include "base/io_buf.h"
+#include "base/logging.h"
+#include "io/io.h"
+#include "server/common.h"
+#include "server/error.h"
+#include "server/journal/types.h"
+#include "server/main_service.h"
+#include "server/transaction.h"
+
+using namespace std;
+
+using nonstd::make_unexpected;
+
+// TODO: Stolen from rdb, unite in common utils
+
+#define SET_OR_RETURN(expr, dest)              \
+  do {                                         \
+    auto exp_val = (expr);                     \
+    if (!exp_val) {                            \
+      VLOG(1) << "Error while calling " #expr; \
+      return exp_val.error();                  \
+    }                                          \
+    dest = exp_val.value();                    \
+  } while (0)
+
+#define SET_OR_UNEXPECT(expr, dest)            \
+  {                                            \
+    auto exp_res = (expr);                     \
+    if (!exp_res)                              \
+      return make_unexpected(exp_res.error()); \
+    dest = std::move(exp_res.value());         \
+  }
```

Comment:
I'm still in favour of moving all the serialization into one folder and share common utils